### PR TITLE
Features2

### PR DESCRIPTION
**Pull request by ckdesir**
Adds many new features to increase functionality of the comment board!

More specifically allows the user the ability to delete comments, sort the comments by either the date they were written or the length, and choose a max number of comments to display. Increases security by preventing possible HTML-injection and cross-site scripting.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/data/Comment.java [hunk 1]
```diff
@@ -8,6 +8,20 @@
  **/
 public class Comment {
 
+  public enum SortType {
```
Comment (chukwuemekagooglecorp) [inline portfolio/src/main/java/com/google/sps/data/Comment.java:11]: Please comment this enum, as well, just like you do with the classes

File: portfolio/src/main/webapp/script-home.js [hunk 2]
```diff
@@ -101,8 +103,33 @@ function makeDiv(comment){
...
-  divOfComment.style.padding='10px';
+  divOfComment.className = 'comment-div'
   return divOfComment;
 }
+
+/**
+ * Returns the URL to the servlet and pdates the query
+ * string based on the max-number-display and sort-comments select.
+ */
+function updateComments() {
```
Comment (Sarsate) [inline portfolio/src/main/webapp/script-home.js:114]: Rename to buildUpdateCommentsURL to make it clear that this function isn't actually modifying comments and only building the request path.

File: portfolio/src/main/webapp/index.html [hunk 3]
```diff
@@ -165,11 +165,35 @@ <h5>What? — May 27, 2020</h5>
...
+    <label for="select-max-numbers">Select maximum number of
+      comments to display:</label>
+    <select id="max-number-display" name="max-numbers"
+      onchange="populateComments()">
+      <option value="5">5</option>
+      <option value="10">10</option>
+      <option value="15" selected>15</option>
+      <option value="20">20</option>
+      <option value="25">25</option>
+      <option value="100">100</option>
```
Comment (Sarsate) [inline portfolio/src/main/webapp/index.html:177]: Curious if you actually tested this ;)

Comment (ckdesir) [reply portfolio/src/main/webapp/index.html:177]: I did up to 20 and thought the rest would work ;i

**Pull request merged**